### DISCUSSION TITLE
posts/Additive-Animation/

### DISCUSSION THREAD

**Discussion by giscus** [ungod/ungod.github.io#7] [Announcements]
# posts/Additive-Animation/

叠加动画常用于战斗受击、射击抖动、AO等。其作用是能同时两个动画的表现，如角色在跑步的时候受击，跑动动画是不能停的同时也要有被击中的抽搐表现；又如角色瞄准的时候射击，射击会因为武器的后坐力让角色躯体发生抖动，而角色也会下意识地维持着瞄准动作……叠加动画几乎占据角色动画逻辑的一大部分，不过似乎很少人讨论有关叠加动画的技术细节。本以UE实现方法为例，意在分析叠加动画的原理和一些细节。

https://ungod.github.io/posts/Additive-Animation/

**Comment by CWolf98** [comment]
作者您好，您的文章启发性很大，但我有一个疑惑希望能得到您的解答：
关于“因为手部的Y轴朝向跟脊椎的旋转轴一样，其Y轴朝向不会受到角色脊椎Yaw的影响”这句话，您的动图中脊椎骨应该是绕着X轴在旋转吧？（我看了下SK_Manny它的脊椎骨是X轴指向下一关节的）
类似的，还有您说的“官方例子中，1主要是产生MeshSpace下的Roll旋转（如图红轴），子骨骼叠加的X方向将不受父骨骼Roll的任意旋转影响，从而产生3的结果”，在官方的例子中base anim应该是绕着Y轴做了旋转。    
 请问是我哪里遗漏了吗？感激不尽

**Reply by ungod** [reply]
感谢你的认真阅读，十分抱歉现在才看到回复。
“因为手部的Y轴朝向跟脊椎的旋转轴一样，其Y轴朝向不会受到角色脊椎Yaw的影响”这里的Y轴指的是手部的Y轴，如图中的绿色轴，其朝向是不会受到角色Root的Yaw影响（或者如你所说的脊椎X轴，不受这个轴的旋转所影响）。
“官方例子中，1主要是产生MeshSpace下的Roll旋转（如图红轴），子骨骼叠加的X方向将不受父骨骼Roll的任意旋转影响，从而产生3的结果”，UE5的SK_Manny角色Y轴确实是FacingDirection，因为DDC工具在转换时会把X轴和Y轴互换。实际上我们习惯说Roll就是绕X，Pitch就是绕Y，Lean动画是Roll动画，而X常指FacingDirection，我在文中补了一个示意图。
